### PR TITLE
null pointer checks are unnecessary: API should never return valid pointers with an error

### PR DESCRIPTION
**Pull request by simonbyrne**

### CODE REVIEW

File: base/libgit2/repository.jl [hunk 1]
```diff
@@ -1,20 +1,17 @@
...
 """
     LibGit2.GitRepo(path::AbstractString)
 
 Opens a git repository at `path`.
 """
 function GitRepo(path::AbstractString)
     repo_ptr_ptr = Ref{Ptr{Void}}(C_NULL)
-    err = ccall((:git_repository_open, :libgit2), Cint,
+    @check ccall((:git_repository_open, :libgit2), Cint,
                 (Ptr{Ptr{Void}}, Cstring), repo_ptr_ptr, path)
```
Comment (nalimilan) [inline base/libgit2/repository.jl]: This breaks alignment of the second line (same below).

File: base/libgit2/repository.jl [hunk 2]
```diff
@@ -1,20 +1,17 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
+
+
+
```
Comment (StefanKarpinski) [inline base/libgit2/repository.jl]: Lots of extra blank lines...

**Review by nalimilan — APPROVED** [review]
Formally looks good to me, I guess you know what you're doing.

**Review by omus — APPROVED** [review]
👍

**Pull request merged**